std/stdio.d

```diff
@@ -4522,70 +4521,70 @@ Initialize with a message and an error code.
     }
 }
 
-extern(C) void std_stdio_static_this()
+// Undocummented but public because std* handles are aliasing it
+ref File makeGlobal(alias handle)()
 {
-    static import core.stdc.stdio;
-    //Bind stdin, stdout, stderr
-    __gshared File.Impl stdinImpl;
-    stdinImpl.handle = core.stdc.stdio.stdin;
-    .stdin._p = &stdinImpl;
-    // stdout
-    __gshared File.Impl stdoutImpl;
-    stdoutImpl.handle = core.stdc.stdio.stdout;
-    .stdout._p = &stdoutImpl;
-    // stderr
-    __gshared File.Impl stderrImpl;
-    stderrImpl.handle = core.stdc.stdio.stderr;
-    .stderr._p = &stderrImpl;
+    static __gshared File.Impl impl;
+    static __gshared File result;
+    static shared bool initialized;
+    if (!initialized)
+    {
+        // Use double-checking because initOnce is inefficient.
```

Comment:
It's not about storing the initializer atomically at all. It's about observing both it's value and the flag value after they've been stored.
It is exactly the same technique inside initOnce already, only initOnce can't count on plain load, it has to do an acquire (which, on some platforms, requires a memory fence).

As for overloads, it also needs a `@nogc` and `nothrow` one. The latter is a bug with `lazy`, for the former we'd need to allocate the mutex without the GC. I'd propose ditching the `lazy` parameter in favor of template callable, but I fear that ship has sailed.